osu.Game/Overlays/Chat/ChannelControl/ControlItem.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Chat.ChannelControl
+{
+    public class ControlItem : OsuClickableContainer
+    {
+        public event Action<Channel>? OnRequestSelect;
+        public event Action<Channel>? OnRequestLeave;
+
+        [Cached]
+        public readonly BindableInt Mentions = new BindableInt();
+
+        [Cached]
+        public readonly BindableBool Unread = new BindableBool();
```

Comment:
Caching dependencies to direct children is generally frowned upon. Preferred style is to either pass down directly for everything that isn't a bindable, and for bindables, it's to declare a matching bindable on the child components and then use `BindTarget` to set up a binding. Examples of this are all over, one is [here](https://github.com/ppy/osu/blob/f81e5e8b992d73411277593b32697d3fc80cf7be/osu.Game/Beatmaps/Drawables/Cards/BeatmapCardNormal.cs#L85).